queries/spark_benchmark.py

```diff
@@ -0,0 +1,238 @@
+import time
+import copy
+from resource import getrusage as resource_usage, RUSAGE_SELF
+from time import time as timestamp
+from pyspark.sql import SQLContext
+from pyspark.context import SparkContext
+from pyspark.sql.session import SparkSession
+import pyspark
+import os
+from collections import defaultdict
+import statistics
+from pyspark import SparkConf
+import sys
+import time
+from statistics import mean
+
+merged_df_path = "../merged"
+parquet_path = "../parquet"
+
+#conf = SparkConf().set("spark.executor.memory","2g").set("spark.driver.memory","2g").setMaster("local")
+#sc = SparkContext("local")
+#sc = SparkContext(conf=conf)
+#sqlContext = SQLContext(sc)
+#spark = SparkSession.builder.master("local[1]").config("spark.driver.memory","2g").config("spark.executor.memory","2g").getOrCreate()
+spark = SparkSession.builder.master("local[1]").config("spark.ui.port","4050").getOrCreate()
+
+
+def unix_time(function, *args, **kwargs):
+    '''Return `real`, `sys` and `user` elapsed time, like UNIX's command `time`
+    You can calculate the amount of used CPU-time used by your
+    function/callable by summing `user` and `sys`. `real` is just like the wall
+    clock.
+    Note that `sys` and `user`'s resolutions are limited by the resolution on
+    the operating system's software clock (check `man 7 time` for more
+    details).
+    '''
+    start_time, start_resources = timestamp(), resource_usage(RUSAGE_SELF)
+    r = function(*args, **kwargs)
+    end_resources, end_time = resource_usage(RUSAGE_SELF), timestamp()
+    return {'return': r,
+            'real': end_time - start_time,
+            'sys': end_resources.ru_stime - start_resources.ru_stime,
+            'user': end_resources.ru_utime - start_resources.ru_utime}
+
+def has_column(df, col):
+    try:
+        df[col]
+        return True
+    except pyspark.sql.utils.AnalysisException:
+        return False
+
+#load the merged parquet into a df
+def load():
+    df = spark.read.parquet(merged_df_path)
+    return df
+
+def analytics(df, iter_num):   
+   # df.groupBy("`id.orig_h`").count().show(df.count(), False
+   df.select("`id.orig_h`").count()
+  
+  #over each individual parquet instead of the merged file
+  #for root, dirs, files in os.walk(parquet_path, topdown = False):
+   #   for name in files:
+    #      df = spark.read.parquet(os.path.join(root, name))
+     #     if has_column(df, "`id.orig_h`"):
+      #        df.select("`id.orig_h`").count()
+
+def search(df, iter_num):
+   origin = sys.stdout
+   df2 = spark.sql("""SELECT *
+    FROM MERGED
+    WHERE `id.orig_h` = '10.128.0.19'
+    ORDER BY ts
+    LIMIT 5
+    """)
+   sys.stdout = open('/dev/null', 'w')
```

Comment:
It looks like a lot of your functions issue these calls to change `sys.stdout` - can you factor this out, for example into your `benchmark` function? That would simplify the code a bit.